packages/main/src/DatePicker.ts

```diff
@@ -217,6 +222,31 @@ type DatePickerInputEventDetail = {
 		},
 	},
 })
+/**
+ * Fired before the value state of the component is updated internally.
+ * The event is preventable, meaning that if it's default action is
+ * prevented, the component will not update the value state.
+ *
+ * @allowPreventDefault
+ * @public
+ * @param {string} valueState The new <code>valueState</code> that will be set.
+ */
+@event<DatePickerValueStateChangeEventDetail>("value-state-change", {
```

Comment:
This event definition does not work. Maybe remove the <DatePickerValueStateChangeEventDetail> part.

Reply:
He has rebase